src/test/java/org/broadinstitute/consent/http/db/LibraryCardDAOTest.java

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.consent.http.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Date;
+import java.util.List;
+import org.postgresql.util.PSQLException;
+import org.apache.commons.lang3.RandomUtils;
+import org.broadinstitute.consent.http.models.LibraryCard;
+import org.junit.Test;
+
+public class LibraryCardDAOTest extends DAOTestHelper {
+
+  @Test
+  public void testInsertLibraryCard() {
+    LibraryCard card = createLibraryCard();
+    List<LibraryCard> all = libraryCardDAO.findAllLibraryCards();
+    assertTrue(all.contains(card));
+  }
+
+  @Test
+  public void testInsertLibraryCardNegative() {
+    Integer userId = createUser().getDacUserId();
+    Integer institutionId = createInstitution().getId();
+    String stringValue = "value";
+    try { 
+        Integer id = libraryCardDAO.insertLibraryCard(0, institutionId, stringValue, stringValue, stringValue, userId, new Date());
+    } catch (Exception e) {
+        assertEquals("23503", ((PSQLException)e.getCause()).getSQLState());
```

Comment:
We should use the constant values instead of string values here if we're expecting this error. I think this one is `PSQLState.FOREIGN_KEY_VIOLATION`. Similar comment in the other assertions.